### PR TITLE
[UI-1306] Structure that allows customer to specify their desired list of Reactions

### PR DESCRIPTION
**Pull request by maciej-adamczyk**

### CODE REVIEW

File: Sources/PubNubChatComponents/Views/Reaction/MessageReactionListView.swift [hunk 1]
```diff
@@ -29,20 +29,22 @@ import UIKit
 import Combine
 
 public class MessageReactionListComponent: UIView {
-  var reactionProvider: ReactionProvider
+  private let provider: ReactionProvider
   private var reactionButtons: [MessageReactionButtonComponent] = []
```
Comment (jguz-pubnub) [inline Sources/PubNubChatComponents/Views/Reaction/MessageReactionListView.swift:33]: Is it possible to make `reactionButtons` as a `let` variable?

File: Sources/PubNubChatComponents/Views/Reaction/MessageReactionListView.swift [hunk 2]
```diff
@@ -28,61 +28,40 @@
...
-  lazy public var astonishedFaceReactionView = MessageReactionButtonComponent(type: .custom)
-  // 😢 crying face U+1F622
-  lazy public var cryingFaceReactionView = MessageReactionButtonComponent(type: .custom)
-  // 🔥 fire U+1F525
-  lazy public var fireReactionView = MessageReactionButtonComponent(type: .custom)
-
-  var currentCount: Int = 0
+public class MessageReactionListComponent: UIView {
+  private let provider: ReactionProvider
+  private var reactionButtons: [MessageReactionButtonComponent] = []
```
Comment (jguz-pubnub) [inline Sources/PubNubChatComponents/Views/Reaction/MessageReactionListView.swift:33]: It it possible to make `reactionButtons` as a `let` variable?

File: Sources/PubNubChatComponents/Views/Reaction/Provider/ReactionProvider+Extensions.swift [hunk 3]
```diff
@@ -0,0 +1,18 @@
+//
+//  ReactionProvider+Extension.swift
+//  PubNubChat
+//
+//  Created by Maciej Adamczyk on 25/08/2022.
+//
+
+import Foundation
+
+extension ReactionProvider {
```
Comment (jguz-pubnub) [inline Sources/PubNubChatComponents/Views/Reaction/Provider/ReactionProvider+Extensions.swift:10]: This extension is short enough so we could keep it in the same file where `ReactionProvider` was defined. But it's also OK to have it in a separate file, I don't mind.

File: Sources/PubNubChatComponents/Views/Reaction/Provider/ReactionProvider.swift [hunk 4]
```diff
@@ -0,0 +1,28 @@
+//
+//  ReactionProvider.swift
+//  PubNubChat
+//
+//  Created by Maciej Adamczyk on 25/08/2022.
+//
+
+import Foundation
+
+public protocol ReactionProvider {
```
Comment (jguz-pubnub) [inline Sources/PubNubChatComponents/Views/Reaction/Provider/ReactionProvider.swift:10]: Do we need any Swift comments above this declaration? What's the purpose of this protocol, what strings are allowed to pass, etc.

File: Sources/PubNubChatComponents/Views/Reaction/MessageReactionListView.swift (hunk 2 above)
Comment (maciej-adamczyk) [reply Sources/PubNubChatComponents/Views/Reaction/MessageReactionListView.swift:33]: I was thinking about making it dynamic, so if we have 2 emoji in message then only 2 buttons are in the list, if we do not have any than list is empty.

**Comment by maciej-adamczyk** [comment]
I am closing this PR, changes are available in 1307 PR

**Pull request closed without merging**